Задача 9 спринта

DariaKasackaja opened the pull request:
Первый pull

Commit:
_

Commit:
LinearLayout+TextView+ImageView заменен на TextView c параметром app:drawableEndCompat

Commit:
Добпалены стили для темной и светлой темы. Спринт 9. Домашнее задание 1.

Commit:
Тема добавлена в AndroidManifest.Поправлена собираемость проекта. Спринт 9. Домашнее задание 1.

Commit:
Тема добавлена в AndroidManifest.Поправлена собираемость проекта. Спринт 9. Домашнее задание 1.

Commit:
Добавлены обработчики кнопок экрана "Настройки". Спринт 9. Домашнее задание 2.

Commit:
Смоделирован экран "Поиск", добавлены обработчики элементов. Спринт 9. Домашнее задание 3.

Commit:
Данные поискового запроса экрана "Поиск" сохраняются при повороте экрана. Спринт 9. Домашнее задание 4.

Commit:
Поправлена иконка кнопки на экране Поиск. Спринт 9. Домашнее задание 4.

anastasianadelshtehel commented on app/src/main/java/com/mifareread/playlistmaker/SearchActivity.kt:
```diff
@@ -1,12 +1,76 @@
+
+        buttonClear.setOnClickListener{ view ->
+            searchEditText.setText("")
+            val inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
```
Управление клавиатурой лучше вынести в отдельный метод

anastasianadelshtehel commented on app/src/main/java/com/mifareread/playlistmaker/SettingsActivity.kt:
```diff
@@ -12,5 +15,31 @@ class SettingsActivity: AppCompatActivity() {
+        buttonSupport.setOnClickListener {
+            val intentSupport = Intent(Intent.ACTION_SENDTO)
+            intentSupport.data = Uri.parse("mailto:")
+            intentSupport.putExtra(Intent.EXTRA_EMAIL, getString(R.string.mail))
```
Интенты лучше вынести в отдельные методы (аналогично для всех)

```
private fun shareApp() {
    val intent = Intent(Intent.ACTION_SEND).apply {
        type = "text/plain"
        putExtra(Intent.EXTRA_TEXT, getString(R.string.share_text))
    }
    startActivity(Intent.createChooser(intent, getString(R.string.share_question)))
}
```

anastasianadelshtehel commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -22,55 +23,22 @@
             android:id="@+id/library"
-            android:layout_width="match_parent"
-            android:layout_height="0dp"
             android:layout_marginTop="16dp"
```
Повторяющиеся значения лучше выносить

anastasianadelshtehel commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -0,0 +1,44 @@
+    <LinearLayout style="@style/BackgroundRectangleStyle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginStart="16dp"
+        android:layout_marginEnd="16dp"
+        android:orientation="horizontal"
+        android:paddingStart="12dp"
+        android:paddingEnd="12dp">
```
Здесь тоже значения должны быть вынесены в ресурсы

anastasianadelshtehel commented:
Нужно поправить экраны в соответствии с версткой
![5276021570134993414](https://github.com/user-attachments/assets/4d1260e9-2a69-4401-b2a0-f2e58463e718)
![5276021570134993415](https://github.com/user-attachments/assets/7410b6cd-3368-4a94-821d-ffc24543bb44)

anastasianadelshtehel commented:
верстка:
![image](https://github.com/user-attachments/assets/033548c6-16e5-4e24-90cf-a99028d7449b)
![image](https://github.com/user-attachments/assets/61ab5f42-b8cf-4342-907d-cc7e18aacd95)

Commit:
Спринт 9. Исправлены замечания 1 ревью:
-управление клаиатурой перенесено в отдельную функцию
-интенты вынесены в отдельные методы
-повторяющиеся значения перенесены в ресурсы
-тулбары в экранах "Поиск" и "Настройки" исправлены
-из EditText убрано подчеркивание
